src/main/java/org/terasology/segmentedpaths/controllers/SegmentCacheSystem.java

```diff
@@ -35,16 +35,14 @@
 @Share(value = SegmentCacheSystem.class)
 public class SegmentCacheSystem extends BaseComponentSystem {
 
-    /**
-    * The map, which stores already constructed segments with their prefab's name as a key.
-    */
+    //A map that stores already constructed segments with their prefab's name as a key.
     private HashMap<String, Segment> segments = new HashMap<>();
 
     /**
-     * Gets a segment based off its prefab.
-     * If a segment was already constructed, returns a segment from the cache, otherwise constructs it from a prefab.
-     * @param prefab which describes a segment.
-     * @return segment from the cache or a freshly constructed one, null if segment's prefab is missing a path-describing component.
+     * Gets a segment based off the given prefab.
+     * If the specified segment has already been constructed, returns it from the cache, otherwise constructs it from the prefab.
+     * @param prefab A prefab that describes the segment.
+     * @return A segment based on the given prefab - either from the cache, or a freshly constructed one, null if the given prefab is missing a path-describing component.
```

Comment:
Now that the rest of it makes sense, I am wondering about the return value - do we have to specify "eitiher from the cache, or a newly constructed one"? Can't we just say `@return A segment based on the given prefab, or null if the given prefab is..`?

Reply:
It will return the path if it finds a matching prefab in the cache else it generates a new one. It's more an if then condition rather then either.

Reply:
so you already defined the behaviour in the description for the return just tell the end user that the function will either return a segment or null if the segment does not have a path descriptor.
```{@link org.terasology.segmentedpaths.components.PathDescriptorComponent}```

Reply:
Please don't ignore my comment :P